pyvista/plotting/mapper.py

```diff
@@ -264,7 +267,10 @@ def scalar_map_mode(self) -> str:
         return vtk_to_pv[self.GetScalarModeAsString()]
 
     @scalar_map_mode.setter
-    def scalar_map_mode(self, scalar_mode: str):
+    def scalar_map_mode(self, scalar_mode: Union[str, FieldAssociation]):
+        if isinstance(scalar_mode, FieldAssociation):
+            scalar_mode = scalar_mode.name
+        scalar_mode = scalar_mode.lower()  # type: ignore
```

Comment:
I'm also wondering about this type ignore. `scalar_mode` is either an `str` (in which case `scalar_mode.lower()` is _also_ an `str`, or `scalar_mode` is something like `FieldAssociation.POINT.name`, which is _also_ `str`.

According to the mypy playground mypy knows the latter as well if I feed it this:
```py
import enum


class FieldAssociation(enum.Enum):
    """Represents which type of vtk field a scalar or vector array is associated with."""

    POINT = 2
    CELL = 1
    NONE = 42
    ROW = 3


var = FieldAssociation.POINT.name
reveal_type(var)  # builtins.str
```
And if I run `mypy pyvista` on the PR branch tip I get the same errors when I remove this type ignore.

So when or how did this line cause a typing issue?

Reply:
I'm not sure why this happening... How mypy behaves is always a mystery to me... I'll try to dig into this but may merge without addressing

Reply:
Actually. I'm not seeing any mypy errors when removing the `# type: ignore`.. are you seeing an error, @adeak?

```
pre-commit run --all-files mypy
mypy.....................................................................Passed
```

But you ran `mypy pyvista`? I did that and don't see this specific issue:

```
(pyvista-dev) ➜  pyvista git:(feat/algorithms) ✗ mypy pyvista
pyvista/core/dataset.py:1761: error: Incompatible return value type (got "ndarray[Any, Any]", expected "pyvista_ndarray")
pyvista/core/composite.py:274: error: Incompatible types in assignment (expression has type "Union[float, ndarray[Any, Any]]", variable has type "float")
pyvista/core/composite.py:276: error: Incompatible types in assignment (expression has type "Union[float, ndarray[Any, Any]]", variable has type "float")
pyvista/ext/plot_directive.py:112: error: Library stubs not installed for "docutils.parsers.rst" (or incompatible with Python 3.9)
pyvista/ext/plot_directive.py:113: error: Library stubs not installed for "docutils.parsers.rst.directives.images" (or incompatible with Python 3.9)
pyvista/ext/plot_directive.py:113: note: Hint: "python3 -m pip install types-docutils"
pyvista/ext/plot_directive.py:113: note: (or run "mypy --install-types" to install all missing stub packages)
pyvista/ext/plot_directive.py:113: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
pyvista/ext/plot_directive.py:292: error: Need type annotation for "plot_context" (hint: "plot_context: Dict[<type>, <type>] = ...")
Found 6 errors in 3 files (checked 91 source files)
```

Reply:
I didn't run the pre-commit hook, I ran mypy manually. But my point is exactly that I can't see any error either way.

Reply:
Oh, then I'll just remove that ` # type: ignore` comment and call it good

Alex had added it in a9392de6d662d05842bab6532f83266dfe3ae8e8 for some reason...

Reply:
Comment removed in 79b545ba1f6e1e756e8c745be8147c2e8b881fd7